src/js/index.js

```diff
@@ -21,15 +21,74 @@ const allowDifficultySelect = shouldAllow => sortingInput.querySelectorAll('.dif
 		node.disabled = !shouldAllow;
 	});
 
+const parameters = {
+	tags: {
+		default: '',
+		getValue: () => selectr.getValue().join(' '),
+		setValue: value => selectr.setValue(value.split(' '))
+	},
+	difficulty: {
+		default: 'all',
+		getValue: () => filterInput.value,
+		setValue: value => {
+			filterInput.value = ['easy', 'medium', 'hard'].includes(value) ?
+				value :
+				'all';
+		}
+	},
+	sort: {
+		default: 'alphabetical',
+		getValue: () => sortParams.sortMethod.toLowerCase(),
+		setValue: value => {
+			sortParams.sortMethod = ['alphabetical', 'difficulty'].includes(value) ?
+				value :
+				'alphabetical';
+		}
+	},
+	order: {
+		default: 'asc',
+		getValue: () => `${sortParams.order.split('SC')[0]}sc`.toLowerCase(),
+		setValue: value => {
+			sortParams.order = value === 'desc' ?
+				'descending' :
+				'ascending';
+		}
+	}
+};
+
 let search;
 let selectr;
+const sortParams = {
+	sortMethod: sortingInput.value.split('_')[0],
+	order: sortingInput.value.split('_')[1]
+};
+
+function updateUrl() {
+	const newSearch = new URLSearchParams(window.location.search);
+	for (const parameter in parameters) {
```

Comment:
If you want to stick with a `for` loop, you want to use [`for ... of {}`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) instead of `for ... in {}` you uses so you can ditch that ugly hasOwnProperty check.